mongodb/src/main/java/com/yahoo/ycsb/db/MongoDbClient.java

```diff
@@ -272,7 +275,8 @@ public int insert(String table, String key,
                 BulkWriteResult result = collection.withWriteConcern(
                         writeConcern).bulkWrite(bulkInserts,
                         new BulkWriteOptions().ordered(false));
-                if (result.getInsertedCount() == bulkInserts.size()) {
+                if (!result.wasAcknowledged()
```

Comment:
Ah, I see. I was interpreting it as a result, rather than an indicator of the write configuration. Thanks for the explanation.